### PR TITLE
Remove unused macros

### LINKED ISSUE #1056: nonsense mix of macros
There are macros in src/dec_macros.h:
* SUB_FIELD_VECTOR_N
* SUB_FIELD_VECTOR_INL

and the same macros in src/spec.h:
* SUB_FIELD_VECTOR_N
* SUB_FIELD_VECTOR_INL

When I need to use src/spec.h/SUB_FIELD_VECTOR_INL macro in dwg.spec, there is an issue.
src/decode.c includes src/dec_macros.h and src/spec.h is included after from src/dwg.spec.

### PR DESCRIPTION
**Pull request by michal-josef-spacek**
Fixes #1056

### CODE REVIEW

File: src/dec_macros.h [hunk 1]
```diff
@@ -1160,22 +1160,6 @@
                              dxf)                                             \
         }                                                                     \
     }
-#define SUB_FIELD_VECTOR_N(o, nam, type, csize, dxf)                          \
```
Comment (rurban) [inline src/dec_macros.h:1163]: No, that's the needed allocating variant

File: src/dec_macros.h [hunk 2]
```diff
@@ -1215,19 +1199,6 @@
         LOG_TRACE ("} [*" #type "]")                                          \
       LOG_POS                                                                 \
     }
-// inlined, with const size and without malloc
-#define SUB_FIELD_VECTOR_INL(o, nam, type, csize, dxf)                        \
-  if (csize > 0)                                                              \
-    {                                                                         \
-      _VECTOR_CHKCOUNT_STATIC (nam, csize, TYPE_MAXELEMSIZE (type), dat)      \
```
Comment (rurban) [inline src/dec_macros.h:1222]: and that's the variant with the bounds check

**Review by rurban — CHANGES_REQUESTED** [review]

**Comment by rurban** [comment]
At first glance this is wrong

**Comment by michal-josef-spacek** [comment]
@rurban As I understand these versions are not used anywhere. Sure?

**Pull request closed without merging**